### PR TITLE
Fix missing /api/v1/apps/verify_credentials endpoint

### PR DESCRIPTION
**Pull request by DataDrivenMD**
[X] Fixes https://github.com/cloudflare/wildebeest/issues/253
[X] Implements validation logic for `POST /api/v1/apps`
[X] Expands test suite for `POST /api/v1/apps` and `GET /api/v1/apps/verify_credentials` endpoints

### CODE REVIEW

**Review by 53jk1 — APPROVED** [review]

**Comment by DataDrivenMD** [comment]
After further testing, I think this PR merits a clarification: this isn't meant to fix **all** 3rd-party app incompatibilities, just those that invoke `GET /api/v1/apps/verify_credentials` during their onboarding process. I'm currently working on a separate fix that I *hope* will fix https://github.com/cloudflare/wildebeest/issues/35

**Comment by jahanson** [comment]
@DataDrivenMD  I tried merging this to my wildebeest instance and I couldn't get it to work on elk.zone. It just spit out json still. Is there anything else you did?

**Comment by DataDrivenMD** [comment]
I'm seeing the same thing. Turns out that there's *another* (separate) Mastodon API incompatibility issue that prevents Elk and Ivory from completing the OAuth flow. I'm working on a fix to that one now, but it's a little more involved because it requires updating type definitions for `MastodonAccount` and `CredentialAccount`. It also involves writing SQL queries to retrieve actual values for several attributes. At the moment, many of the required attributes are hard-coded to static values, which seems to break OAuth flow for anything other than the official Mastodon iOS App.

File: functions/api/v1/apps.ts [hunk 1]
```diff
@@ -23,7 +23,24 @@ export async function handleRequest(db: D1Database, request: Request, vapidKeys:
 		return new Response('', { status: 400 })
 	}
 
-	const body = await readBody<AppsPost>(request)
+	const body: AppsPost = await readBody<AppsPost>(request)
+
+  // Parameter validation according to https://github.com/mastodon/mastodon/blob/main/app/lib/application_extension.rb
+  if ((body.client_name === undefined) || (body.client_name?.trim() === '')) {
+    return new Response('Unprocessable entity: client_name cannot be an empty string', { status: 422 })
```
Comment (xtuc) [inline functions/api/v1/apps.ts:30]: Can you return the errors as JSON?  You can reuse the logic in `backend/src/errors/index.ts`

File: backend/test/mastodon/apps.spec.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import { strict as assert } from 'node:assert/strict'
+import * as apps from 'wildebeest/functions/api/v1/apps'
+import * as verify_app from 'wildebeest/functions/api/v1/apps/verify_credentials'
+import { CredentialApp } from 'wildebeest/functions/api/v1/apps/verify_credentials'
+import { VAPIDPublicKey } from 'wildebeest/backend/src/mastodon/subscription'
+import { generateVAPIDKeys } from 'wildebeest/backend/test/mastodon.spec'
+
+describe('Mastodon APIs', () => {
+	describe('/apps', () => {
+    test('POST /apps registers client', async () => {
```
[the remark this answers is not in the document]
Comment (DataDrivenMD) [reply backend/test/mastodon/apps.spec.ts:12]: Will earmark time later today to do this + return the erros as JSON + add CORS headers

**Comment by xtuc** [comment]
Is my understanding correct that Elk and Ivory are more deeply incompatible with Wildebeest because of their ID UUID vs Integer issue?  I can take a look at removing the hardcoded values in the Oauth flow

**Comment by DataDrivenMD** [comment]
> Is my understanding correct that Elk and Ivory are more deeply incompatible with Wildebeest because of their ID UUID vs Integer issue? I can take a look at removing the hardcoded values in the Oauth flow

@xtuc Sorry for the delayed response. That's the primary incompatibility with Ivory based on my DMs with the Ivory team. 

> I tried merging this to my wildebeest instance and I couldn't get it to work on elk.zone. It just spit out json still. Is there anything else you did?

@jahanson Elk had another issue going on, which has since been fixed－ I'm able to log into my Wildebeest instance from my [Elk instance](https://elk.fedified.com), and you're welcome to try logging into your Wildebeest instance from my Elk instance to try to narrow down the source of the problem, if you'd like.

**Comment by DataDrivenMD** [comment]
Sorry, @xtuc not sure why the linting is still failing PR checks. Would appreciate your help getting past this roadblock.

**Comment by xtuc** [comment]
@DataDrivenMD  I can't update your branch directly (I believe the `allow edits from maintainers` option isn't checked) but here's a commit to fix linting https://github.com/cloudflare/wildebeest/commit/c44d552d080dd4e633efcfe7c4e48d9e427f1f67

**Comment by DataDrivenMD** [comment]
Seems like I have to upgrade my organization plan to enable that option. Will look into it. In any case, I manually applied your commit (thank you!) and that seems to have done the trick thanks for your help.

**Review by xtuc — APPROVED** [review]
Thanks

**Pull request merged**